eventnotifier/slack/slack.go

```diff
@@ -96,105 +107,174 @@ func (n *Notifier) GetNotifierName() string {
 		return fmt.Errorf("slack notifier: %w", err)
 	}
 
-	n.logger.Debugw("notifier: sending notifier event to slack", "disruptionName", dis.Name, "eventType", event.Type, "message", bodyText)
+	n.logger.Debugw("notifier: sending notifier event to slack", "message", slackMsg.BodyText)
 
 	return nil
 }
 
-func (n *Notifier) sendMessageToChannel(userInfo authv1.UserInfo, slackChannel, headerText string, headerBlock *slack.HeaderBlock, disruptionBlocks []*slack.TextBlockObject, bodyBlock *slack.SectionBlock) {
+func (n *Notifier) sendMessageToUserChannel(obj client.Object, slackMsg slackMessage, logger *zap.SugaredLogger) error {
+	objKind := obj.GetObjectKind().GroupVersionKind().Kind
+
+	emailAddr, err := mail.ParseAddress(slackMsg.UserInfo.Username)
+	if err != nil {
+		logger.Warnw("invalid user info email in %s %s: %w", objKind, obj.GetName(), err)
+
+		return nil
+	}
+
+	p1, err := n.client.GetUserByEmail(emailAddr.Address)
+	if err != nil {
+		logger.Warnw("user not found", "userAddress", emailAddr.Address, "error", err)
+
+		return nil
+	}
+
+	return n.sendMessageToChannel(p1.ID, slackMsg)
+}
+
+func (n *Notifier) sendMessageToChannel(slackChannel string, slackMsg slackMessage) error {
 	userName := infoNotAvailable
-	if userInfo.Username != "" {
-		userName = userInfo.Username
+	if slackMsg.UserInfo.Username != "" {
+		userName = slackMsg.UserInfo.Username
 	}
 
 	_, _, err := n.client.PostMessage(slackChannel,
-		slack.MsgOptionText(headerText, false),
-		slack.MsgOptionUsername("Disruption Status Bot"),
+		slack.MsgOptionText(slackMsg.HeaderText, false),
+		slack.MsgOptionUsername(slackMsg.UserName),
 		slack.MsgOptionIconURL("https://upload.wikimedia.org/wikipedia/commons/3/39/LogoChaosMonkeysNetflix.png"),
 		slack.MsgOptionBlocks(
-			headerBlock,
+			slackMsg.HeaderBlock,
 			slack.NewDividerBlock(),
-			slack.NewSectionBlock(nil, append(disruptionBlocks, slack.NewTextBlockObject("mrkdwn", "*Author:*\n"+userName, false, false)), nil),
+			slack.NewSectionBlock(nil, append(slackMsg.InfoBlocks, slack.NewTextBlockObject("mrkdwn", "*Author:*\n"+userName, false, false)), nil),
 			slack.NewDividerBlock(),
-			bodyBlock,
+			slackMsg.BodyBlock,
 		),
 		slack.MsgOptionAsUser(true),
 	)
+
+	return err
+}
+
+func (n *Notifier) buildSlackMessage(obj client.Object, event corev1.Event, notifType types.NotificationType, reporting *v1beta1.Reporting, logger *zap.SugaredLogger) slackMessage {
+	headerText := utils.BuildHeaderMessageFromObjectEvent(obj, event, notifType)
+	headerBlock := slack.NewHeaderBlock(slack.NewTextBlockObject("plain_text", headerText, false, false))
+	bodyText := utils.BuildBodyMessageFromObjectEvent(obj, event, true)
+	bodyBlock := slack.NewSectionBlock(slack.NewTextBlockObject("mrkdwn", bodyText, false, false), nil, nil)
+	infoBlocks := n.buildSlackBlocks(obj, notifType, reporting)
+
+	var (
+		userInfo authv1.UserInfo
+		err      error
+	)
+
+	switch d := obj.(type) {
+	case *v1beta1.Disruption:
```

Comment:
It's completely arbitrary 😇 
You prefere with if statements?

``` golang
var (
	userInfo authv1.UserInfo
	err      error
)

if d, ok := obj.(*v1beta1.Disruption); ok {
	userInfo, err = d.UserInfo()
} else if d, ok := obj.(*v1beta1.DisruptionCron); ok {
	userInfo, err = d.UserInfo()
}

if err != nil {
	logger.Errorw("unable to retrieve user info", "error", err)
}
```

And also, I'm forced to check the type because the `UserInfo()` method is only linked to the `Disruption` and `DisruptionCron` and not for the `Object`